### PR TITLE
target index added

### PR DESCRIPTION
**Pull request by leandrodzp**
* List all targets from a user functionality added

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -7,6 +7,18 @@
     namespace :v1, defaults: { format: :json } do
       devise_scope :user do
         resources :targets, only: :create
+        # namespace :users do
```
Comment (letiesperon) [inline config/routes.rb]: Why the commented code?

File: spec/requests/api/v1/users/targets/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    end
+
+    it 'returns success' do
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'returns all user\'s targets' do
+      expect(json[:targets].length).to eq(user.targets.count)
+    end
+  end
```
Comment (letiesperon) [inline spec/requests/api/v1/users/targets/index_spec.rb]: Separate contexts with blank line

File: spec/requests/api/v1/users/targets/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  context 'when user has no targets' do
+    it 'returns a nil array' do
+      get api_v1_user_targets_path(user), headers: auth_headers, as: :json
+      expect(json[:targets].length).to eq(0)
```
Comment (letiesperon) [inline spec/requests/api/v1/users/targets/index_spec.rb]: to be empty

File: spec/requests/api/v1/users/targets/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+describe 'POST api/v1/users/:user_id/targets', type: :request do
+  let(:user)    { create(:user, confirmed_at: Time.zone.now) }
+
+  context 'when user has at least 1 target' do
+    let!(:targets) { create_list(:target, 8, user: user) }
```
Comment (letiesperon) [inline spec/requests/api/v1/users/targets/index_spec.rb]: 8 is a bit much considering they are inserted to the db one by one 😅

File: config/routes.rb [hunk 5]
```diff
@@ -7,6 +7,18 @@
...
+        #   resources :targets, only: :index
```
Comment (gonzaM) [inline config/routes.rb]: Remove commented code.

File: spec/requests/api/v1/users/targets/index_spec.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    it 'returns success' do
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'returns all user\'s targets' do
+      expect(json[:targets].length).to eq(user.targets.count)
+    end
+  end
+  context 'when user has no targets' do
+    it 'returns a nil array' do
```
Comment (gonzaM) [inline spec/requests/api/v1/users/targets/index_spec.rb]: Change to `'returns empty array'`

File: spec/requests/api/v1/users/targets/index_spec.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      get api_v1_user_targets_path(user), headers: auth_headers, as: :json
+      expect(json[:targets].length).to eq(0)
+    end
+  end
```
Comment (gonzaM) [inline spec/requests/api/v1/users/targets/index_spec.rb]: You should check that the information is correct. At least, check that the returned targets ids are the same as the existing ones.

**Comment by leandrodzp** [comment]
Done 💃

**Comment by gonzaM** [comment]
Leti comments are fixed, and she already approved the PR, so I will merge the PR.

**Pull request merged**